src/Process/ProcessFactory.php

```diff
@@ -0,0 +1,120 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Process;
+
+use function func_get_args;
+use Infection\Console\Util\PhpProcess;
+use Infection\Utils\VersionParser;
+use PackageVersions\Versions;
+use Symfony\Component\Process\Process;
+use function version_compare;
+
+/**
+ * @internal
+ *
+ * This factory is only used for dealing with the SymfonyProcess 4.4 deprecation. It can be safely
```

Comment:
With current implementation, after we upgrade to `^4.0 || ^5.0` dependencies as we 

1. will have to literally do a revert of this commit 
2. and remove version comparison

On the other side, leaving the code as is (in `master`), we will only need to 

1. remove version comparison

So isn't it a complication?

Reply:
It is a complication for a performance improvement. But the question is whether we plan to bump those dependencies soon or not

Reply:
I'd rather see us upgrading to `^4.0 || ^5.0` tbh

Reply:
> It is a complication for a performance improvement

Here is what we have running that code 1 time instead of N (number of Mutants)

```php
$start = microtime(true);

for ($i = 0; $i < 1; $i++) {
    $symfonyProcessVersion = $versionParser->parse(Versions::getVersion('symfony/process'));

    version_compare($symfonyProcessVersion, '4.4.0', '<');
}

var_dump(microtime(true) - $start);
die;
```

Result: `0.0033740997314453s` for 2000 calls

1 call is `8.9883804321289E-5s`.

We can neglect these multiple calls to version parsers&comparers and not introduce this complexity.

> I'd rather see us upgrading to ^4.0 || ^5.0 tbh

Well, I'm not strongly against, but is this version comparison a really reason to bump dependencies? IMO doesn't worth it.